ql/src/java/org/apache/hadoop/hive/ql/parse/LoadSemanticAnalyzer.java

```diff
@@ -294,6 +295,14 @@ private void analyzeLoad(ASTNode ast) throws SemanticException {
       throw new SemanticException(ErrorMsg.DML_AGAINST_VIEW.getMsg());
     }
     if (ts.tableHandle.isNonNative()) {
+      // launch a tez job
+      StorageFormatDescriptor ss = ts.tableHandle.getStorageHandler().supportsLoadData(ts.tableHandle.getTTable());
+      if (ss != null) {
```

Comment:
Can we simply do (see MergeSemanticAnalyzer)
````
StorageFormat format = new StorageFormat(conf);
format.processStorageFormat("blabla");
````

Reply:
Could have done, but the inner logic is same, we have to check if native table is iceberg and then write.format and all constants aren't accessible from here, show some string hardcoding.

For Append we have to any way go to StorageHandler to iceberg module to execute the append API's